state/watcher.go

```diff
@@ -2081,21 +2093,35 @@ func (u *Unit) WatchLXDProfileUpgradeNotifications() (StringsWatcher, error) {
 type instanceCharmProfileDataWatcher struct {
```

Comment:
Suggest renaming to something more generic now that a collection can be specified and isn't hard coded.  As is the field watched.